packages/radio-player/README.md

```diff
@@ -0,0 +1,120 @@
+# \<radio-player>
+
+A Radio Player that displays closed captioning and allows searching.
+
+![Radio Player](./assets/img/radio-player.png "Radio Player Demo")
+
+## Installation
+```bash
+yarn add @internetarchive/radio-player
+```
+
+## Usage
+```js
+// radio-player.js
+import RadioPlayer from '@internetarchive/radio-player';
+export default RadioPlayer;
+```
+
+```html
+<!-- index.html -->
+<script type="module">
+  import './radio-player.js';
+
+  const quickSearchTerms = [];
+
+  const audioSource = new AudioSource(
+    'https://ia803005.us.archive.org/30/items/BBC_Radio_2_20190502_180000/BBC_Radio_2_20190502_180000.mp3',
+    'audio/mpeg',
+  );
+
+  const radioConfig = new RadioPlayerConfig(
+    'Voice of America',
+    '7:00pm',
+    './logo.jpg',
+    './waveform.png',
+    [audioSource],
+    quickSearchTerms,
+  );
+
+  const transcriptEntries: TranscriptEntryConfig[] = [...];
+
+  const transcriptConfig = new TranscriptConfig(transcriptEntries);
+</script>
+
+<style>
+  radio-player {
+    line-height: 1.5rem;
+    color: white;
+
+    --timeColor: white;
+    --timeColumnWidth: 3rem;
+    --transcriptHeight: 200px;
+
+    --autoScrollButtonFontColor: black;
+    --autoScrollButtonBackgroundColor: white;
+
+    --normalTextColor: gray;
+    --activeTextColor: white;
+    --searchResultInactiveBorderColor: gray;
+    --searchResultActiveBorderColor: green;
+
+    --trackColor: black;
+    --trackBorder: 1px solid white;
+  }
+</style>
+
+<radio-player
+  .config=${radioConfig}
+  .transcriptConfig=${transcriptConfig}
+  @searchRequested=${this.searchRequested}
```

Comment:
Can these listener methods be the ones used by default rather than explicitly passing each in?

Reply:
From the look of it, the README is suggesting passing the radio player instance's methods into itself to attach as event handlers. Instead, can we attach them as defaults in the render method and accept optional overrides for each? I haven't used lit-element other than tinkering with it so far, so I'm not sure if this is easily accomplished.

Reply:
Ohhh I think I see what you're saying. I think this might be where the lines of demarcation get a bit fuzzy between the "radio player" and its host. The radio player itself doesn't handle any searching (or really any data-type events), it just sends the `searchRequested`, `searchCleared`, etc events up to its host and then the host is responsible for executing the search and rebuilding the transcript / search results to pass back down into the radio player. This is what the `radio-player-controller` is doing. It's acting as the bridge between the host and the radio player.

I will add a bit more to this readme, though, to make it more clear what's going on here.

Did I answer your question?

Reply:
Ah, yeah. I thought those methods were already present on the component, but if they're supplied by whatever controller you're creating to interact with the view layer it makes sense.